packages/material-ui/src/ObjectFieldTemplate/ObjectFieldTemplate.tsx

```diff
@@ -1,16 +1,49 @@
 import React from 'react';
 
 import Grid from '@material-ui/core/Grid';
+import Button from '@material-ui/core/Button';
+import AddIcon from '@material-ui/icons/Add';
+
 import { makeStyles } from '@material-ui/styles';
+import { UiSchema, ObjectFieldTemplateProps, utils } from '@rjsf/core';
+import { JSONSchema7 } from 'json-schema';
+
+const { getUiOptions } = utils;
+
+type AddButtonProps = {
+  onClick: (e?: any) => void,
+  disabled: boolean
+};
 
-import { ObjectFieldTemplateProps } from '@rjsf/core';
+const AddButton = (props: AddButtonProps) => (
+  <Button {...props} color="secondary">
+    <AddIcon /> Add Item
+  </Button>
+);
 
 const useStyles = makeStyles({
   root: {
     marginTop: 10,
   },
 });
 
+const canExpand = function canExpand(formData: object, schema: JSONSchema7, uiSchema: UiSchema) {
```

Comment:
This function seems to be very similar to the [canExpand function in @rjsf/antd](https://github.com/rjsf-team/react-jsonschema-form/blob/f378583a27365bc2ae5cd99897d7385abd58949a/packages/antd/src/templates/ObjectFieldTemplate/index.js#L81). Can you instead move this function to @rjsf/core utils.js and then import it from there?